### PR TITLE
More strict on what is considered a builtin - Fixes #16

### LINKED ISSUE #16: Prettier Plugin not respecting config when called via CLI
**Your Environment**

-   **Prettier version**: 2.6.2
-   **node version**: 16.13.2
-   **package manager**: yarn@1.22.18
-   **IDE**: VSCode 1.67.1

**Describe the bug**

In our repo, when we use a CLI command to rewrite all files, it successfully executes prettier, and maybe some of the plugin, but not all the configuration is preserved. Effectively it looks like it ignores the `importOrder` setting from the config.

**To Reproduce**

(Using the config provided below)

With this test file, already formatted via prettier, but without the prettier-plugin-sort-imports installed.

<details><summary>testFile.ts</summary>

```ts
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/naming-convention */

import { DateTime } from "luxon";
import { Inject, Service } from "typedi";

import EmailService from "./email";
import {
	CareNotesDatabase,
	OfficeStaffDatabase,
	ShiftsDatabase,
	OfficeReport,
	OfficeStaffContactInfo,
	OfficeStaffRoles,
	AgencyConfigDatabase,
} from "../database";

import { EmailRecipient, Shift, TriggerProp } from "../types";
import SMSService from "./sms";
import { logger } from "@hca/logger";
import MESSAGES, { MESSAGE_TRIGGER_PROPS, MESSAGE_TYPES } from "../constants/messages";
import snakecase from "lodash.snakecase";

export interface OfficeReportMessageData {
	careNoteDocId: string;
}
```

</details>

Run `yarn prettier --write testFile.ts` to see:
<details><summary>CLI-formatted:</summary>

```ts
/* eslint-disable @typescript-eslint/no-unused-vars */

/* eslint-disable @typescript-eslint/naming-convention */
import MESSAGES, { MESSAGE_TRIGGER_PROPS, MESSAGE_TYPES } from "../constants/messages";

import snakecase from "lodash.snakecase";
import { DateTime } from "luxon";
import { Inject, Service } from "typedi";

import { logger } from "@hca/logger";

import {
	AgencyConfigDatabase,
	CareNotesDatabase,
	OfficeReport,
	OfficeStaffContactInfo,
	OfficeStaffDatabase,
	OfficeStaffRoles,
	ShiftsDatabase,
} from "../database";
import { EmailRecipient, Shift, TriggerProp } from "../types";
import EmailService from "./email";
import SMSService from "./sms";

export interface OfficeReportMessageData {
	careNoteDocId: string;
}
```

</details>

Trigger IDE to invoke prettier to auto-format to see:
<details><summary>Editor Formatted</summary>

```ts
/* eslint-disable @typescript-eslint/no-unused-vars */

/* eslint-disable @typescript-eslint/naming-convention */
import snakecase from "lodash.snakecase";
import { DateTime } from "luxon";
import { Inject, Service } from "typedi";

import { logger } from "@hca/logger";

import MESSAGES, { MESSAGE_TRIGGER_PROPS, MESSAGE_TYPES } from "../constants/messages";
import {
	AgencyConfigDatabase,
	CareNotesDatabase,
	OfficeReport,
	OfficeStaffContactInfo,
	OfficeStaffDatabase,
	OfficeStaffRoles,
	ShiftsDatabase,
} from "../database";
import { EmailRecipient, Shift, TriggerProp } from "../types";
import EmailService from "./email";
import SMSService from "./sms";

export interface OfficeReportMessageData {
	careNoteDocId: string;
}
```

</details>

**Expected behavior**

I would expect the IDE-prettier-invoked output to be the final result, since that matches my `importOrder` configuration.

Given that it does make changes to the layout when invoking it via CLI with the plugin installed, I think the plugin isn't respecting the options from the configuration file when invoked via CLI. I have verified that prettier is indeed discovering the plugin via CLI `--support-info` and the above technique.

**Configuration File (cat .prettierrc, prettier.config.js, .prettier.js)**

```js
module.exports = {
	overrides: [
		{
			// prettier will strip newlines out of package.json files unless you tell it to use the json parser
			files: ["package.json", "**/package.json"],
			options: { parser: "json" },
		},
	],
	printWidth: 100,
	semi: true,
	singleQuote: false, // Single quotes are common in text-strings
	trailingComma: "all", // Improves refactoring / minimizes git-conflicts
	tabWidth: 2,
	useTabs: true,
	// @ianvs/prettier-plugin-sort-imports
	importOrder: ["^@hca", "^[.]+"],
	importOrderBuiltinModulesToTop: true,
	importOrderCaseInsensitive: true,
	importOrderParserPlugins: ["typescript", "jsx", "decorators-legacy"],
	importOrderSeparation: true,
	importOrderSortSpecifiers: true,
	// @prettier/plugin-xml
	xmlWhitespaceSensitivity: "ignore",
};
```

**Error log**
N/A

**Contribute to @ianvs/prettier-plugin-sort-imports**

-   [x] I'm willing to fix this bug 🥇

### PR DESCRIPTION
**Pull request by fbartho**
Relates to #16

Looks like a combination of config options are conflicting?

Not sure why it triggers on CLI but not in IDE for me, but the run_spec is able to reproduce it in this repo.

### CODE REVIEW

**Comment by IanVS** [comment]
@fbartho I've invited you as a collaborator, so you can make branches without a fork, to make it easier to collaborate.  I'm taking a look into this now.  I updated my project at work to the latest version, and running prettier in the cli does not throw any errors, so at least it doesn't seem like something that broke in the latest release.

**Comment by IanVS** [comment]
This appears to be related to `importOrderBuiltinModulesToTop`.  Setting that to `false` makes this test case pass.

**Comment by IanVS** [comment]
`constants` is one of the builtin node modules.  The regex needs to be more explicit.  I should be able to crank out a fix, thanks for pointing this out!

**Comment by blutorange** [comment]
How about [changing this](https://github.com/IanVS/prettier-plugin-sort-imports/blob/54b3cf5c84c3762ec70e93ab197243cc73b8f3c8/src/constants.ts#L19) to

> `^(${builtinModules.join("|")}|${builtinModules.map(module => 'node:'+module).join("|")})$`

That resolves to

> ^(_http_agent|_http_client|_http_common|_http_incoming|_http_outgoing|_http_server|_stream_duplex|_stream_passthrough|_stream_readable|_stream_transform|_stream_wrap|_stream_writable|_tls_common|_tls_wrap|assert|assert/strict|async_hooks|buffer|child_process|cluster|console|constants|crypto|dgram|diagnostics_channel|dns|dns/promises|domain|events|fs|fs/promises|http|http2|https|inspector|module|net|os|path|path/posix|path/win32|perf_hooks|process|punycode|querystring|readline|repl|stream|stream/consumers|stream/promises|stream/web|string_decoder|sys|timers|timers/promises|tls|trace_events|tty|url|util|util/types|v8|vm|wasi|worker_threads|zlib|node:_http_agent|node:_http_client|node:_http_common|node:_http_incoming|node:_http_outgoing|node:_http_server|node:_stream_duplex|node:_stream_passthrough|node:_stream_readable|node:_stream_transform|node:_stream_wrap|node:_stream_writable|node:_tls_common|node:_tls_wrap|node:assert|node:assert/strict|node:async_hooks|node:buffer|node:child_process|node:cluster|node:console|node:constants|node:crypto|node:dgram|node:diagnostics_channel|node:dns|node:dns/promises|node:domain|node:events|node:fs|node:fs/promises|node:http|node:http2|node:https|node:inspector|node:module|node:net|node:os|node:path|node:path/posix|node:path/win32|node:perf_hooks|node:process|node:punycode|node:querystring|node:readline|node:repl|node:stream|node:stream/consumers|node:stream/promises|node:stream/web|node:string_decoder|node:sys|node:timers|node:timers/promises|node:tls|node:trace_events|node:tty|node:url|node:util|node:util/types|node:v8|node:vm|node:wasi|node:worker_threads|node:zlib)$

**Comment by IanVS** [comment]
Yep, exactly.  Although I used a non-capturing group.  I also simplified down the test case a bit.

File: src/constants.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export const chunkTypeOther = 'other';
  * where the not matched imports should be placed
  */
 export const THIRD_PARTY_MODULES_SPECIAL_WORD = '<THIRD_PARTY_MODULES>';
-export const BUILTIN_MODULES = builtinModules.join('|');
+export const BUILTIN_MODULES = `^(?:${builtinModules.join('|')})$`;
```
Comment (blutorange) [inline src/constants.ts:19]: Careful, this is missing support for node protocol imports, e.g. `require("node:fs")`

```js
`^(${builtinModules.join("|")}|${builtinModules.map(module => 'node:'+module).join("|")})$`
```

Comment (blutorange) [reply src/constants.ts]: Or perhaps a bit simpler:

```js
`^(node:)?(${builtinModules.join("|")})$`
```

Comment (IanVS) [reply src/constants.ts]: I should have run all the tests before pushing it up.  :)

I tweaked the regex a bit to allow an optional `node:` prefix.

**Comment by IanVS** [comment]
Oh this is interesting.  I guess that `node:fs/promises` wasn't valid in node 12.  I'll adjust the tests a bit.

**Comment by IanVS** [comment]
I guess the other option is to stop supporting node 12.  It's EOL, after all.  I propose that we remove it from our tests, at least.  What do you think, @blutorange, @fbartho?

**Comment by fbartho** [comment]
I'm certainly supportive of only explicitly supporting N-1 on stable node release lines. -- Like maybe it works on node-12, but we shouldn't try to maintain support for it!

File: src/constants.ts (hunk 1 above)
Comment (fbartho) [reply src/constants.ts]: You said you tweaked the regex @IanVS, but I admit, I don't understand how that tweak works? -- or the tweak hasn't been pushed up yet?

Here's my playground: https://regex101.com/r/d7kjI0/1

Comment (fbartho) [reply src/constants.ts]: (I see the fix now)

**Comment by fbartho** [comment]
The PR is green! Please let me know if you want me to strip out the test case, or move it somewhere, or if this PR is ready to go!

**Review by IanVS — APPROVED** [review]
Looks good to me, thanks both of you.

**Comment by IanVS** [comment]
Released in https://github.com/IanVS/prettier-plugin-sort-imports/releases/tag/v3.4.2.

**Pull request merged**